talend-component-maven-plugin/src/it/web/.jenkins/Jenkinsfile

```diff
@@ -0,0 +1,322 @@
+    volumes:
+      - name: docker
+        hostPath: { path: /var/run/docker.sock }
+      - name: efs-jenkins-component-runtime-m2
+        persistentVolumeClaim:
+              claimName: efs-jenkins-component-runtime-m2
+      - name: dockercache
+        hostPath: { path: /tmp/jenkins/tdi/docker }
+    imagePullSecrets:
+      - name: talend-registry
+"""
+
+/**
+ * Credentials
+ */
+
+final def nexusCredentials = usernamePassword(
+    credentialsId: 'nexus-artifact-zl-credentials',
+    usernameVariable: 'NEXUS_USER',
+    passwordVariable: 'NEXUS_PASSWORD')
+
+final def tenantId_Rd = string(
+    credentialsId: 'component-runtime-api-tenant-id-rd',
+    variable: 'componentRuntimeApiTenantId_Rd'
+)
+
+/**
+ * Options
+ */
+
+final String serverPort = "8081"
+final String serverLog = "component_server.log"
+final String serverTimeout = "1"
+
+// Default value for parameters
+String runtimeVersion = "default"
+String apiTesterEnv = "component_runtime_ci"
+String fileToRun = "tcomp_approved"
+String tenantInstance = "eu"
+/**
+ * The pipeline
+ */
+pipeline {
+  /**
+   * agent
+   */
+  agent {
+    kubernetes {
+      label podLabel
+      yaml podConfiguration
+      defaultContainer stageDefaultContainer
+    }
+  }
+  /**
+   * parameters
+   */
+  parameters {
+    string(
+      name: 'CYCLE',
+      defaultValue: 'default',
+      description: 'Test Cycle Id filled by Test Orchestrator, it is only important for TTO.'
+    )
+    string(
+      name: 'RUNTIME_VERSION',
+      defaultValue: 'default',
+      description: 'Choose the component-runtime.version to be used from Nexus. Keep default to use the pom version (1.45.0 minimum) or an other ex: 1.45.0-SNAPSHOT'
+    )
+    choice(
+      name: 'PROJECT_NAME' ,
+      choices: ['tcomp_approved', 'tcomp_draft'] ,
+      description: 'Select test project to execute.'
+    )
+  }
+  /**
+   * environment
+   */
+  environment {
+    MAVEN_SETTINGS = "${WORKSPACE}/.jenkins/settings.xml"
+    DECRYPTER_ARG = "-Dtalend.maven.decrypter.m2.location=${env.WORKSPACE}/.jenkins/"
+    MAVEN_OPTS = [
+      "-Dmaven.artifact.threads=128",
+      "-Dorg.slf4j.simpleLogger.showDateTime=true",
+      "-Dorg.slf4j.simpleLogger.showThreadName=true",
+      "-Dorg.slf4j.simpleLogger.dateTimeFormat=HH:mm:ss",
+      "-Dtalend-image.layersCacheDirectory=/root/.dockercache"
+    ].join(' ')
+
+    APP_ID = '579232'
+    TALEND_REGISTRY = "artifactory.datapwn.com"
+    TESTCONTAINERS_HUB_IMAGE_NAME_PREFIX = "artifactory.datapwn.com/docker-io-remote/"
+
+    ROOT_PATH = "${WORKSPACE}"
+    SCRIPTS_PATH = "${WORKSPACE}/talend-component-maven-plugin/src/it/web/.jenkins/scripts"
+    TESTS_PATH = "${WORKSPACE}/talend-component-maven-plugin/src/it/web/test"
+    CONNECTORS_PATH = "${WORKSPACE}/connectors"
+    TESTED_CONNECTOR_PATH = "${CONNECTORS_PATH}/azure/azure-dls-gen2"
+  }
+  /**
+   * options
+   */
+  options {
+    buildDiscarder(logRotator(artifactNumToKeepStr: '30', numToKeepStr: '30'))
```

Comment:
that's a lot, no?

Reply:
I did remove ROOT_PATH.
CONNECTORS_PATH and TESTED_CONNECTOR_PATH may be merged later with the usage of the demo connector